### ISSUE TITLE
Sound effects no longer work under Windows 7 (DirectX)

### ISSUE THREAD

**Issue by MichaelDePiazzi** [mono/MonoGame#4045]
PR #4010 has caused a regression - Sound effects no longer work under Windows 7 (for the DirectX build).

Further investigation reveals that SoundEffect(.XAudio).InitializeSoundEffect was swallowing an exception about being unable to load "xaudio2_8.dll". I found a post on the SharpDX forums discussing this issue (http://sharpdx.org/forum/5-api-usage/3334-xaudio2-unable-to-load-dll-xaudio2-8-dll). Long story short - It seems you need to use the DirectX11 SharpDX libraries for Windows 7. The DirectX11.1/11.2 libraries don't seem to be fully supported under Windows 7.

I did a quick hack to prove this by copying the SharpDX 2.6.3 DirectX11 libraries in. I also had to revert the changes made to SoundEffect.XAudio.cs after I did this to get it compiling. Sound effects started working again with these changes.

I'm more than happy to make a pull request for this. But I wasn't sure whether to use the SharpDX "DirectX11-net40" or the "DirectX11-Signed-net40" libraries though...

**Comment by tomspilman** [comment]
> But I wasn't sure whether to use the SharpDX "DirectX11-net40" or the "DirectX11-Signed-net40" libraries though...

@mattleibow ???

**Comment by MichaelDePiazzi** [comment]
@tomspilman I took a quick look at whether the previous SharpDX libraries being used for Windows Desktop were the signed (i.e. strong named) version.

The ones committed by mattleibow are the signed version (which makes sense as the SharpDX packages from NuGet are apparently all using the signed versions, and this is where he got it from). But the previous commit of the libraries (v2.5.1) were the standard version (I believe RayBatts committed these - https://github.com/Mono-Game/MonoGame.Dependencies/pull/6 / https://github.com/Mono-Game/MonoGame.Dependencies/pull/7).

So I guess that doesn't really help answer the question on which to use. And I can't really find any pros or cons to using either in MonoGame. But my opinion is that the standard libraries should be fine as they were being used prior to this. And there doesn't seem to be any particular reason that I can find to start using the signed versions in MonoGame.

**Comment by mattleibow** [comment]
@tomspilman @MichaelDePiazzi It is probably better to use the signed versions everywhere, this allows for other libraries to be signed. For example, I know the CocosSharp team signs their library, and they had problems with the standard dlls.

**Comment by mattleibow** [comment]
@MichaelDePiazzi The code you had to revert, was it the:

```csharp
const int deviceId = 0; 
var deviceDetails = Device.GetDeviceDetails(deviceId);    
Speakers = deviceDetails.OutputFormat.ChannelMask;
```

@tomspilman to support both the latest features, we could probably try something on the lines of this:
```csharp
try
{
    const int deviceId = 0; 
    var deviceDetails = Device.GetDeviceDetails(deviceId);    
    Speakers = deviceDetails.OutputFormat.ChannelMask;
}
catch (EntryPointNotFoundException ex)
{
    Speakers = (Speakers)MasterVoice.ChannelMask;
}
```

This should be OK as this is in the static constructor and only is done once. What do you think?

**Comment by MichaelDePiazzi** [comment]
@mattleibow I just reverted all changes you made to SoundEffect.XAudio.cs to get things working.

The DirectX11 version uses an int for the deviceId in the MasteringVoice constructor, whereas DirectX11.2 uses a string. MasteringVoice doesn't have the ChannelMask property either in DirectX11. Also, the debug code still works under the DirectX11 version so is probably worth leaving in.

The try/catch wouldn't work as it would cause a compilation error due to the slightly different APIs. So the conditional compilation which was originally being used is still required.

**Comment by mattleibow** [comment]
@MichaelDePiazzi I get you...

**Comment by mattleibow** [comment]
@tomspilman @MichaelDePiazzi I have created a PR reverting the requirement for DX11.1, it now works with DX11.

**Comment by MichaelDePiazzi** [comment]
Thanks @mattleibow !

**Comment by MichaelDePiazzi** [comment]
This is now fixed by PR #4057.
